dll/win32/shlwapi/appcompat.c

```diff
@@ -0,0 +1,443 @@
+typedef struct tagAPPCOMPATINFO
+{
+    PCSTR pszAppName;
+    PCSTR pszAppVersion;
+    DWORD dwCompatFlags;
+} APPCOMPATINFO, *PAPPCOMPATINFO;
+static APPCOMPATINFO g_appCompatInfo[] =
+{
+    { "wpwin7.exe", NULL, SHACF_CONTEXTMENU | SHACF_CORELINTERNETENUM },
+    { "prwin70.exe", NULL, SHACF_CONTEXTMENU | SHACF_CORELINTERNETENUM },
+    { "ps80.exe", NULL, SHACF_CONTEXTMENU | SHACF_CORELINTERNETENUM | SHACF_OLDREGITEMGDN },
+    { "qpw.exe", MAJOR_VER_ONLY "7", SHACF_CONTEXTMENU },
+    { "qfinder.exe", NULL, SHACF_CORELINTERNETENUM | SHACF_OLDREGITEMGDN },
+    { "pfim80.exe", NULL, SHACF_CONTEXTMENU | SHACF_CORELINTERNETENUM  | SHACF_OLDREGITEMGDN },
+    { "ua80.exe", NULL, SHACF_CONTEXTMENU | SHACF_CORELINTERNETENUM  | SHACF_OLDREGITEMGDN },
+    { "pdxwin32.exe", NULL, SHACF_CONTEXTMENU | SHACF_CORELINTERNETENUM  | SHACF_OLDREGITEMGDN },
+    { "sitebuilder.exe", NULL, SHACF_CONTEXTMENU | SHACF_CORELINTERNETENUM  | SHACF_OLDREGITEMGDN },
+    { "hotdog4.exe", NULL, SHACF_DOCOBJECT },
+    { "rnaapp.exe", NULL, SHACF_CONTEXTMENU },
+    { "pdexplo.exe", MAJOR_VER_ONLY "2", SHACF_CONTEXTMENU | SHACF_MYCOMPUTERFIRST },
+    { "pdexplo.exe", MAJOR_VER_ONLY "1", SHACF_CONTEXTMENU | SHACF_MYCOMPUTERFIRST },
+    { "pdexplo.exe", MAJOR_VER_ONLY "3", SHACF_MYCOMPUTERFIRST | SHACF_OLDREGITEMGDN },
+    { "sizemgr.exe", MAJOR_VER_ONLY "3", SHACF_CORELINTERNETENUM | SHACF_OLDREGITEMGDN },
+    { "smartctr.exe", "96.0", SHACF_CONTEXTMENU },
+    { "wpwin8.exe", NULL, SHACF_CORELINTERNETENUM | SHACF_OLDREGITEMGDN },
+    { "prwin8.exe", NULL, SHACF_CORELINTERNETENUM | SHACF_OLDREGITEMGDN },
+    { "ue32.exe", "2.00.0.0", SHACF_OLDREGITEMGDN },
+    { "pp70.exe", NULL, SHACF_LOADCOLUMNHANDLER },
+    { "pp80.exe", NULL, SHACF_LOADCOLUMNHANDLER },
+    { "ps80.exe", NULL, SHACF_OLDREGITEMGDN },
+    { "abcmm.exe", NULL, SHACF_LOADCOLUMNHANDLER },
+    { "qpw.exe", MAJOR_VER_ONLY "8", SHACF_CORELINTERNETENUM | SHACF_OLDREGITEMGDN | SHACF_ANSIDISPLAYNAMES },
+    { "coreldrw.exe", MAJOR_VER_ONLY "7", SHACF_OLDREGITEMGDN },
+    { "filler51.exe", NULL, SHACF_OLDREGITEMGDN },
+    { "autorun.exe", "4.10.1998", SHACF_ANSI },
+    { "autorun.exe", "4.00.950", SHACF_ANSI },
+    { "powerpnt.exe", MAJOR_VER_ONLY "8", SHACF_WIN95SHLEXEC },
+    { "msmoney.exe", "7.05.1107", SHACF_WIN95SHLEXEC },
+    { "soffice.exe", MAJOR_VER_ONLY "5", SHACF_STAROFFICE5PRINTER },
+    { "wpwin9.exe", MAJOR_VER_ONLY "9", SHACF_CORELINTERNETENUM },
+    { "qpw.exe", MAJOR_VER_ONLY "9", SHACF_CORELINTERNETENUM },
+    { "prwin9.exe", MAJOR_VER_ONLY "9", SHACF_CORELINTERNETENUM },
+    { "dad9.exe", MAJOR_VER_ONLY "9", SHACF_CORELINTERNETENUM },
+};
+
+// Window class name and compatibility flags
+typedef struct tagWNDCOMPATINFO
+{
+    PCSTR pszLengthAndClassName;
+    DWORD dwCompatFlags;
+} WNDCOMPATINFO, *PWNDCOMPATINFO;
+static WNDCOMPATINFO g_wndCompatInfo[] =
+{
+    // The first byte indicates the string length
+    { "\x09" "bosa_sdm_", 0x1000100 },
```

Comment:
And the second element (0x1000100) is a combination of which flags?

Reply:
Fixed.